src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -972,26 +970,27 @@ void append_probe_realpath(const pal::string_t& path, std::vector<pal::string_t>
 std::vector<host_option> fx_muxer_t::get_known_opts(bool exec_mode, host_mode_t mode, bool get_all_options)
 {
     std::vector<host_option> known_opts = { { _X("--additionalprobingpath"), _X("<path>"), _X("Path containing probing policy and assemblies to probe for") } };
-    if (get_all_options || exec_mode || mode == host_mode_t::split_fx || mode == host_mode_t::standalone)
+    if (get_all_options || exec_mode || mode == host_mode_t::split_fx || mode == host_mode_t::apphost)
     {
-        known_opts.push_back({ _X("--depsfile"), _X("<path>"),  _X("Path to <application>.deps.json file") });
-        known_opts.push_back({ _X("--runtimeconfig"), _X("<path>"), _X("Path to <application>.runtimeconfig.json file") });
+        known_opts.push_back({ _X("--depsfile"), _X("<path>"), _X("Path to <application>.deps.json file")});
+        known_opts.push_back({ _X("--runtimeconfig"), _X("<path>"), _X("Path to <application>.runtimeconfig.json file")});
+        known_opts.push_back({ _X("--dotnet-path"), _X("<path>"), _X("Path to the installed Shared Framework to use to run the application.")});
     }
 
-    if (get_all_options || mode == host_mode_t::muxer)
+    if (get_all_options || mode == host_mode_t::muxer || mode == host_mode_t::apphost )
     {
-        known_opts.push_back({ _X("--fx-version"), _X("<version>"), _X("Version of the installed Shared Framework to use to run the application.") });
-        known_opts.push_back({ _X("--roll-forward-on-no-candidate-fx"), _X(""), _X("Roll forward on no candidate shared framework is enabled") });
-        known_opts.push_back({ _X("--additional-deps"), _X("<path>"), _X("Path to additonal deps.json file") });
+        // If mode=host_mode_t::apphost, these are only used when the app is framework-dependent.
```

Comment:
How will this ever get into this code for the apphost?  If you invoke it with no args, it should run the app.  If you invoke it with anything like `--help`, it should pass `--help` on to the app.

I don't think these are possible for apphost.

Reply:
The code just above this supports `--depsfile` and `--runtimeconfig` options for apphost. I just verified against 2.0.6 that they can be used with apphost. The changes here also add the other three valid options below (for framework-dependent apphost anyway)

Reply:
What if the app wanted to take an argument named the same thing?  It would seem impossible to support.

It seems incorrect to process arguments at all in the apphost.

Reply:
There is a chicken-egg situation currently where we parse the args before we determine if the app is self-contained or framework-dependent, so it's a bit of refactoring (if we want) to remove the 3 extra args from parsing when under apphost + self-contained.

Reply:
I don't plan on changing 2.0.x functionality; the original code does include standalone\apphost:
```
if (get_all_options || exec_mode || mode == host_mode_t::split_fx || mode == host_mode_t::standalone) 
{
    known_opts.push_back({ _X("--depsfile"), _X("<path>"),  _X("Path to <application>.deps.json file") }); 
    known_opts.push_back({ _X("--runtimeconfig"), _X("<path>"), _X("Path to <application>.runtimeconfig.json file") }); 
```

and this "feature" (if so) makes sense to also include the other 3 (which would only do anything in framework-dependent).